bears/general/AnnotationBear.py

```diff
@@ -1,6 +1,81 @@
+        """
+        Finds out all the positions of strings and comments in a file.
+        The Bear searches for valid comments and strings and yields their
+        ranges as SourceRange objects in HiddenResults.
+
+        :param language:        Language to be whose annotations are to be
+                                searched.
+        :param language_family: Language family whose annotations are to be
+                                searche.
+        :return:                HiddenResult with content conatining
+                                a tuple of tuples of the form
+                                (annotation_start, annotation_end).
+        """
+        text = ''.join(file)
+        lang_dict = LanguageDefinition(language, language_family)
+
+        # Strings
+        strings = dict(lang_dict["string_delimiters"])
+        strings.update(lang_dict["multiline_string_delimiters"])
+        strings_found = find_start_end(text, strings, escape=True)
+        # multiline Comments
+        comments_found = find_start_end(
+                                text,
+                                dict(lang_dict["multiline_comment_delimiters"]))
+        # single-line Comments
+        comments_found += find_singleline_comments(
+                                          text,
+                                          list(lang_dict["comment_delimiter"]))
+
+        matches_found = strings_found + comments_found
+        # Remove Nested
+        matches_found = tuple(filter(
+                              lambda arg: not in_ranges(matches_found, arg),
+                              matches_found))
+        # Yield results
+        for annotation_found in [strings_found, comments_found]:
+            position_range = tuple(AnnotationBear.get_range(text,
+                                                            filename,
+                                                            matches_found,
+                                                            annotation_found))
+            yield HiddenResult(self, position_range)
+
+    @classmethod
+    def get_range(cls, text, filename, matches_found, annotation_found):
+        """
+        yields position of valid annotations as SourceRange objects.
+
+        :param text:             A string with all the contents of file.
+        :param filename:         Name of file.
+        :param matches_found:    All the valid strings and comments found.
+        :param annotation_found: All the positions of the given annotation.
+
+        :return:                 SourceRange Objects of all the valid positions
+                                 of the annotation.
+            """
+        for position in set(matches_found) & set(annotation_found):
+            yield SourceRange.from_values(
+                    filename,
+                    start_line=calc_line_col(text,
+                                             position[0])[0],
+                    start_column=calc_line_col(text,
```

Comment:
something like:

```
kwargs = {}
(kwargs['start_line'],
 kwargs['start_column']) = calc_line_col(text, position[0])
(kwargs['end_line'],
 kwargs['end_column']) = calc_line_col(text, position[1])
# and using **kwargs
```
would be easier to read and faster !
you are actually parsing the file 4 times while you can do it with only 2 times. (Call `calc_line_col` as less as possible)

In fact because of this, I would encourage you to modify the `calc_line_col` to take in a list of positions, then parse the file and return a list of tuples of (line, col)
You can assume they're sorted (and Raise an exception if not) or sort them internally. This would increase efficiency.